code/game/objects/items/motion_detector.dm

```diff
@@ -1,353 +1,147 @@
-
-#define MOTION_DETECTOR_LONG 0
-#define MOTION_DETECTOR_SHORT 1
-
-#define MOTION_DETECTOR_HOSTILE ""
-#define MOTION_DETECTOR_FRIENDLY "_friendly"
-#define MOTION_DETECTOR_DEAD "_dead"
-#define MOTION_DETECTOR_FUBAR "_fubar" //i.e. can't be revived. Might have useful gear to loot though!
-
-
-/obj/effect/detector_blip
+/obj/effect/edge_blip
 	icon = 'icons/Marine/marine-items.dmi'
-	icon_state = "detector_blip"
-	layer = BELOW_FULLSCREEN_LAYER
+	plane = ABOVE_HUD_PLANE
+	/// A friendly/hostile identifier
 	var/identifier = MOTION_DETECTOR_HOSTILE
-	var/edge_blip = FALSE
-	var/turf/center
-	var/turf/target_turf
-
-/obj/effect/detector_blip/friendly
-	icon_state = "detector_blip_friendly"
-	identifier = MOTION_DETECTOR_FRIENDLY
 
-/obj/effect/detector_blip/dead
-	icon_state = "detector_blip_dead"
-	identifier = MOTION_DETECTOR_DEAD
+/obj/effect/edge_blip/Initialize(mapload, identifier, mob/operator, screen_pos_x, screen_pos_y, direction = SOUTH)
+	. = ..()
+	if(!operator?.client)
+		return INITIALIZE_HINT_QDEL
+	screen_loc = "[screen_pos_x],[screen_pos_y]"
+	operator.client.screen += src
+	src.identifier = identifier
+	setDir(direction)
+	update_icon()
+	addtimer(CALLBACK(src, .proc/remove_blip, operator), 1 SECONDS)
 
-/obj/effect/detector_blip/fubar
-	icon_state = "detector_blip_fubar"
-	identifier = MOTION_DETECTOR_FUBAR
+/// Remove the blip from the operator screen
+/obj/effect/edge_blip/proc/remove_blip(mob/operator)
+	operator.client.screen -= src
+	qdel(src)
 
-/obj/effect/detector_blip/Destroy()
-	center = null
-	target_turf = null
-	return ..()
+/obj/effect/edge_blip/update_icon_state()
+	icon_state = "edge_blip_[identifier]"
 
-/obj/effect/detector_blip/update_icon_state()
-	icon_state = "detector_blip[edge_blip ? "_dir" : ""][identifier]"
+/obj/effect/close_blip
+	plane = ABOVE_HUD_PLANE
+	var/image/blip_image
 
-/obj/item/motiondetector
+/obj/effect/close_blip/Initialize(mapload, identifier, mob/operator)
+	. = ..()
+	if(!operator?.client)
+		return INITIALIZE_HINT_QDEL
+	blip_image = image('icons/Marine/marine-items.dmi', src, "close_blip_[identifier]")
+	blip_image.layer = BELOW_FULLSCREEN_LAYER
+	operator.client.images |= blip_image
+	addtimer(CALLBACK(src, .proc/remove_blip, operator), 1 SECONDS)
+
```

Comment:
Why dont blips get pooled into a list on the motion detector so you have to use one timer for alll the blips instead of one for each